### PR TITLE
Add ability to set NODE_ENV for npm_package

### PR DESCRIPTION
**Pull request by jeffbyrnes**
## Description

Adds the ability to set `NODE_ENV` for the `npm_package` resource.

Important to note that setting `--production` overrides `NODE_ENV`.

## Issues Resolved

#195

## Check List

- [x] All tests pass.
- [x] New functionality includes testing.
- [x] New functionality has been documented in the README if applicable

### CODE REVIEW

**Comment by MarkGibbons** [comment]
FYI, I'm going to get the GitHub pipeline to run test instances before doing much of anything else.

**Comment by kitchen-porter** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Please include a CHANGELOG entry.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by MarkGibbons** [comment]
Could you add a test to verify that npm_package installed the package as expected?

File: test/cookbooks/test/templates/package.json [hunk 1]
```diff
@@ -10,9 +10,13 @@
...
   },
   "dependencies": {
+    "config": "^1.26.2",
     "koa": "^1.1.0"
   },
   "devDependencies": {
     "gulp": "^3.9.0"
+  },
+  "scripts": {
+    "postinstall": "touch $NODE_ENV"
```
Comment (jeffbyrnes) [inline test/cookbooks/test/templates/package.json:20]: @MarkGibbons so this ½ of the test I wrote to ensure this works. As part of the `postinstall`, it attempts to a file named the same as `$NODE_ENV`.

File: test/integration/source/default.rb [hunk 2]
```diff
@@ -6,4 +6,8 @@
   describe command('npm -v') do
     its('exit_status') { should eq 0 }
   end
+
+  describe file '/home/random/staging' do
+    it { should exist }
```
Comment (jeffbyrnes) [inline test/integration/source/default.rb:11]: @MarkGibbons and here is the other half of the test (which you can see in multiple test files), asserting that the file exists, thus ensuring that `$NODE_ENV` was successfully passed to `npm_package`.

**Review by ramereth — APPROVED** [review]

**Comment by jeffbyrnes** [comment]
It took two years, but here we are! 🎉

**Pull request merged**